packages/paima-sdk/paima-mw-core/src/helpers/general.ts

```diff
@@ -1,7 +1,30 @@
+import { ConfigNetworkType, GlobalConfig } from '@paima/utils';
 import { getWeb3 } from '../state.js';
+import assertNever from 'assert-never';
 
 export async function getBlockNumber(): Promise<number> {
-  return await getWeb3().then(web3 => web3.eth.getBlockNumber());
+  const config = await GlobalConfig.mainConfig();
+  const networkType = config[1].type;
```

Comment:
I think `const [_, mainConfig]` is a bit more intuitive than using `config[1]` in multiple places

Reply:
I think that's better too, it's just that it also generates a 

```
6:10  warning  '_' is assigned a value but never used  @typescript-eslint/no-unused-vars
```

and it's annoying to have to add an exception all the time.

Reply:
to be honest I regret making this thing returning an array instead of an object, and I was going to change it, but I didn't want to do it here since the PR already has a lot of changes.